pkgs/tools/misc/z/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "z-${version}";
+  version = "1.11";
+
+  src = fetchFromGitHub {
+    owner = "rupa";
+    repo = "z";
+    rev = "v${version}";
+    sha256 = "13zbgkj6y0qhvn5jpkrqbd4jjxjr789k228iwma5hjfh1nx7ghyb";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    install -Dm 644 z.sh $out/share/z.sh
```

Comment:
For shell integration something like this would be helpful:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/fzf/default.nix#L44